announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -694,11 +701,42 @@ else if (view.equals(VIEW_MODE_PUBLIC))
 							{
 								messages = getMessagesPublic(site, channel, null, true, state, portlet);
 							}
+							else if (view.equals(VIEW_MODE_BYROLE))
+							{
+								messages = getMessagesByRoles(site, channel, null, true, state, portlet);
+							}
 						}
 						else
 						{
 							messages = getMessages(channel, null, true, state, portlet);
 						}
+						List<AnnouncementWrapper> finalMessages = new ArrayList<>();
```

Comment:
I think all of this behavior should be inside the 'if (view.equals(VIEW_MODE_BYROLE))' logic, am I right?

Reply:
Should this logic be inside 'getMessagesByRoles' ?

Reply:
No, because the getMessagesByRole it is only when the view by role is selected. But this code should be excute everytime